### PR TITLE
OC-7712 Administrative Edit Data in Enketo Forms.

### PR DESCRIPTION
**Pull request by jkeremian**
Administrative Edit Data in Enketo Forms.

### CODE REVIEW

File: web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java [hunk 1]
```diff
@@ -153,34 +152,34 @@ else if ("itemdata".equalsIgnoreCase(entityType)) {
...
+                formUrl = enketoUrlService.getEditUrl(contextHash, subjectContext, currentStudy.getOid(), null, null, flavor);
             } else {
-                request.setAttribute(DataEntryServlet.INPUT_EVENT_CRF_ID, String.valueOf(idb.getEventCRFId()));
-                request.setAttribute(DataEntryServlet.INPUT_SECTION_ID, String.valueOf(ifmb.getSectionId()));
-
+                formUrl = enketoUrlService.getInitialDataEntryUrl(contextHash, subjectContext, currentStudy.getOid(), flavor);
             }
+
+            request.setAttribute(EnketoFormServlet.FORM_URL, formUrl);
+            // } else {
```
Comment (yogi-oc) [inline web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java]: remove this code

File: web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java [hunk 2]
```diff
@@ -153,34 +152,34 @@ else if ("itemdata".equalsIgnoreCase(entityType)) {
...
+            // Cache the subject context for use during xform submission
+            PFormCache cache = PFormCache.getInstance(context);
+            PFormCacheSubjectContextEntry subjectContext = new PFormCacheSubjectContextEntry();
+            subjectContext.setStudySubjectOid(ssb.getOid());
+            subjectContext.setStudyEventDefinitionId(seb.getStudyEventDefinitionId());
+            subjectContext.setOrdinal(seb.getSampleOrdinal());
+            subjectContext.setCrfVersionOid(crfVersion.getOid());
+            subjectContext.setUserAccountId(ub.getId());
+            String contextHash = cache.putSubjectContext(subjectContext);
+            String flavor = "-query";
```
Comment (yogi-oc) [inline web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java]: Can you define a constant for this?

Comment (jkeremian) [inline web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java]: Done in the new pull request

File: web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java (hunk 1 above)
Comment (jkeremian) [inline web/src/main/java/org/akaza/openclinica/control/managestudy/ResolveDiscrepancyServlet.java]: Done in the next pull request

**Pull request merged**